backend/apps/onboarding/services/nudge_engine.py

```
@@ -0,0 +1,136 @@
+                    }
+                }
+            )
+        except Exception as e:
+            logger.error(f"WebSocket nudge failed: {e}")
+
+        logger.info(f"Nudge sent to {journey.user.username}: {nudge_type}")
+
+    def _is_dropping_off(self, journey: OnboardingJourney) -> bool:
+        """Check if journey is dropping off."""
+        # No activity for 7 days
+        last_event = JourneyEvent.objects.filter(
```

Ruff (0.15.21)

[error] Undefined name `JourneyEvent`

(F821)